src/DGmethods/test/Sphere/advection_sphere.jl

```diff
@@ -0,0 +1,290 @@
+        nothing
+    end
+
+    step = [0]
+    mkpath("vtk")
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1000) do (init=false)
+        outprefix = @sprintf("vtk/advection_sphere_%dD_mpirank%04d_step%04d",
+                             3, MPI.Comm_rank(mpicomm), step[1])
+        @printf(io, "----\n")
+        @printf(io, "doing VTK output =  %s\n", outprefix)
+        DGBalanceLawDiscretizations.writevtk(outprefix, Q, spacedisc, ("ρ", ))
+        step[1] += 1
+        nothing
+    end
+    #------------Set Callback Info--------------------------------#
+
+    #Perform Time-Integration
+    solve!(Q, lsrk; timeend=timeend, callbacks=(cbinfo, cbmass, cbvtk))
+
+    # Print some end of the simulation information
+    if integration_testing
+        error = euclidean_distance(Q, Qe) / norm(Qe)
+        Δmass = abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+        @info @sprintf """Finished
+        error = %.16e
+        Δmass = %.16e
+        """ error Δmass
+    end
+
+    #return diagnostics
+    return (error, Δmass)
+end
+#}}} Main
+
+#{{{ Run Script
+function run(mpicomm, Nhorizontal, Nvertical, N, timeend, DFloat, dt, ArrayType)
+    Rrange=range(DFloat(1); length=Nvertical+1, stop=2)
+    topl = StackedCubedSphereTopology(mpicomm,Nhorizontal,Rrange; boundary=(0,0))
+    (error, Δmass) = main(mpicomm, DFloat, topl, N, timeend, ArrayType, dt)
+end
+#}}} Run Script
+
+#{{{ Run Program
+using Test
+let
+    MPI.Initialized() || MPI.Init()
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    if MPI.Comm_rank(mpicomm) == 0
+        ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+        loglevel = ll == "DEBUG" ? Logging.Debug :
+        ll == "WARN"  ? Logging.Warn  :
+        ll == "ERROR" ? Logging.Error : Logging.Info
+        global_logger(ConsoleLogger(stderr, loglevel))
+    else
+        global_logger(NullLogger())
+    end
+
+    #Perform Integration Testing for three different grid resolutions
+    if integration_testing
+        timeend = 1
+        numelem = (2, 2) #(Nhorizontal,Nvertical)
+        N = 4
+        ArrayType = Array
+        dt=1e-2*5 #stable dt for N=4 and Ne=5
+
+        expected_error = Array{Float64}(undef, 3) # h-refinement levels lvl
+        expected_error[1] = 1.5684355102513664e-01 #Ne=2
+        expected_error[2] = 8.8549044843092145e-03 #Ne=4
+        expected_error[3] = 2.2163001051913196e-04 #Ne=8
+        lvls = length(expected_error)
+
+        for DFloat in (Float64,) #Float32)
+            err = zeros(DFloat, lvls)
+            mass= zeros(DFloat, lvls)
+            for l = 1:lvls
+                Nhorizontal = 2^(l-1) * numelem[1]
+                Nvertical   = 2^(l-1) * numelem[2]
+                dt=dt/Nhorizontal
+                nsteps = ceil(Int64, timeend / dt)
+                dt = timeend / nsteps
+                @show(Nhorizontal,Nvertical,N,dt,nsteps)
```

Comment:
I'd suggest removing the `@show`. If you want to show this data, put it inside an `@info` call which will be more MPI friendly.